src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,86 @@
+package com.internet.shop.web.filters;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebFilter("/*")
+public class AuthorizationFilter implements Filter {
+    private static final String USER_ID = "user_id";
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+    private Map<String, List<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/admin/actions", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/users/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/users/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/users/orders", List.of(Role.RoleName.ADMIN));
```

Comment:
what does this controller do? what is the difference between it, `/admin/orders/all` and `/users/orders`?

Reply:
admin/users/orders allows admin at the all users page go to the selected user orders. It is additional controller that I implemented few PRs ago instead of controller that should show orders of user with ID = 1 (I misunderstood the task). 
I had conversation about it with Dima in last PR, and now with you, so the best decision is to remove it, to avoid menthors confusing :)
users/orders shows logged user orders
admin/orders/all shows all orders of all users

Ok, I delete admin/users/orders and close this question.